packages/story-editor/src/components/header/buttons/update.js

```diff
@@ -75,38 +108,45 @@ function UpdateButton({ hasUpdates = false, forceIsSaving = false }) {
   // then only if there are new changes or the story has meta-boxes – as these
   // can update without us knowing it.
   const isEnabled = !isSaving && !isUploading && (hasNewChanges || hasUpdates);
-  let text;
-  switch (status) {
-    case 'publish':
-    case 'private':
-      text = __('Update', 'web-stories');
-      break;
-    case 'future':
-      text = __('Schedule', 'web-stories');
-      break;
-    default:
-      text = __('Save draft', 'web-stories');
-      return (
-        <Tooltip title={text} hasTail>
-          <Button
-            variant={BUTTON_VARIANTS.SQUARE}
-            type={BUTTON_TYPES.QUATERNARY}
-            size={BUTTON_SIZES.SMALL}
-            onClick={() => saveStory({ status: 'draft' })}
-            disabled={!isEnabled}
-            aria-label={text}
-          >
-            <Icons.FloppyDisk />
-          </Button>
-        </Tooltip>
-      );
+
+  if ('pending' === status) {
+    return (
+      <PlainButton
+        text={__('Save as pending', 'web-stories')}
+        disabled={!isEnabled}
+        onClick={() => saveStory()}
+      />
+    );
+  }
+
+  if ('draft' === status) {
+    return (
+      <PlainButton
+        text={__('Save draft', 'web-stories')}
+        disabled={!isEnabled}
+        onClick={() => saveStory({ status: 'draft' })}
+      />
+    );
   }
 
+  // Offset the date by one minute to accommodate for network latency.
+  const hasFutureDate = isAfter(
+    subMinutes(toDate(date, getOptions()), 1),
+    toDate(new Date(), getOptions())
+  );
+
+  const text =
+    hasFutureDate && !status === 'private'
+      ? __('Schedule', 'web-stories')
+      : __('Update', 'web-stories');
+
   return (
     <ButtonWithChecklistWarning
       text={text}
       onClick={() => saveStory()}
       disabled={isSaving || isUploading}
+      isUploading={isUploading}
```

Comment:
Should it be disabled for other media processes as well?

Reply:
+1

Reply:
Worth noting that this is not doing anything new, just fixing a bug.

`ButtonWithChecklistWarning` accepts an `isUploading` prop:

https://github.com/google/web-stories-wp/blob/f22db1e7ddc307a07fabc020c536eb6f8d8a8904/packages/story-editor/src/components/header/buttons/buttonWithChecklistWarning.js#L111-L115

In one place the prop is passed as expected:

https://github.com/google/web-stories-wp/blob/f22db1e7ddc307a07fabc020c536eb6f8d8a8904/packages/story-editor/src/components/header/buttons/publish.js#L111-L117

But in this component here it was simply forgotten.

Any change in behavior beyond that warrants its own ticket.